### PR TITLE
Feature/solution-fetching-all-assignments

### PR DESCRIPTION
**Pull request by w-disaster**
Changes:
- Solve all button
- Memoization for solve all
- `DpllEnumerator` -> impl which returns all assignments, `DpllFinder` -> impl  which returns the first assignment, if available, with the possibility to search for another.
- `Solution`'s `print` changed
- Added `Status` to `Solution` to choose to plot the next button: `COMPLETED` -> no next button, `PARTIAL` -> yes next button.

### CODE REVIEW

File: src/main/scala/satify/update/solver/Solver.scala [hunk 1]
```diff
@@ -51,17 +57,17 @@ object Solver:
   /** Private implementation of [[Solver]] */
   private case class DpllSolver(converter: Converter) extends Solver:
 
-    import DpllSolverMemoize.runDpll
+    override def solveAll(cnf: CNF): Solution = enumerate(cnf)
 
-    override def solve(cnf: CNF): Solution = runDpll(cnf)
+    override def solve(cnf: CNF): Solution = DpllFinder.dpll(cnf)
 
     override def solve(exp: Expression): Solution = solve(converter.convert(exp))
```
Comment (Mala1180) [inline src/main/scala/satify/update/solver/Solver.scala:64]: - `solveAll` name maybe is not really clear, we'll search for a better name
- We need another overloaded method to take in input also an `Expression` (like normal `solve`)

**Review by Mala1180 — APPROVED** [review]
Good job, nice modeling!

File: src/main/scala/satify/update/solver/Solver.scala (hunk 1 above)
Comment (w-disaster) [reply src/main/scala/satify/update/solver/Solver.scala:64]: Why not?
I'm going to make the second change, thanks for the clarification.

**Review by paga16-hash — APPROVED** [review]
Top!

**Pull request merged**